src/api/records.test.tsx

```diff
@@ -0,0 +1,228 @@
+import { Record, RecordRow } from '../app.types';
+import {
+  testRecords,
+  testRecordRows,
+  hooksWrapperWithProviders,
+  getInitialState,
+} from '../setupTests';
+import axios from 'axios';
+import { renderHook, waitFor } from '@testing-library/react';
+import { useRecordCount, useRecordsPaginated } from './records';
+import { PreloadedState } from '@reduxjs/toolkit';
+import { RootState } from '../state/store';
+
+const dataResponsesEqual = (x?: RecordRow[], y?: RecordRow[]): boolean => {
+  if (!x || !y) return false;
+  if (x.length !== y.length) return false;
+
+  for (let i = 0; i < x.length; i++) {
+    const xRow = x[i];
+    const yRow = y[i];
+
+    const xKeys = Object.keys(xRow);
+    const yKeys = Object.keys(yRow);
+
+    for (let i = 0; i < xKeys.length; i++) {
+      if (xKeys[i] !== yKeys[i]) return false;
+    }
+
+    if (xRow.timestamp !== yRow.timestamp) return false;
+    if (xRow.shotnum !== yRow.shotnum) return false;
+    if (xRow.activeArea !== yRow.activeArea) return false;
+    if (xRow.activeExperiment !== yRow.activeExperiment) return false;
```

Comment:
If we iterate over all the keys - isn't it unnecessary to check if the 4 values at the end are the same?

Reply:
So basically, line 26 is iterating to make sure the key names of both objects match.

Lines 29-32 check the _content_ of those fields match. But I've been unable to check the content of the channels in each record row match. I think it's to do with additional fields being added in the response or something happens during the base64 encoding.

This is the whole reason I even made this function instead of just using
```
expect(JSON.stringify(result)).toEqual(JSON.stringify(expected));
```
The above returns an error that the two strings parse to the same thing or something like that (but test fails)

Reply:
If you get the `serializes to the same string` error - can't you solve that by using `toEqual`/`toStrictEqual` or something? Normally that error is caused by using `toBe` in the wrong place.

Reply:
I did try quite hard to use the right comparison (plus plenty of googling) but didn't seem to budge. Since I haven't touched this in a bit, I'll have another go at it again soon before I resolve this convo just to make sure I'm not missing something, because that would be highly preferable to my solution here